Add outdoor air mixer in terminal unit of the VRF-FluidTCtrl model

rongpengzhang opened the pull request:
The VRF-FluidTCtrl model was developed for the TU without OA mixer. If the OA mixer exists, the terminal unit will not handle the load introduced by OA. In this fix, the algorithm was improved to handle the case with OA Mixer in TU. Since the TU of VRF-FluidTCtrl model implements VAV fan and allows varying airflow rate to manipulate the coil capacity, this fix needs to handle the coupling effect between the OA Mixer and VAV Coil simulation.

rongpengzhang committed:
temp

rongpengzhang committed:
Preliminary investigation of the problem

rongpengzhang committed:
Investigate the codes for operational mode determination

rongpengzhang committed:
Activate ControlVRF for VRF-FluidTCtrl model

rongpengzhang committed:
Refine the codes to determine operational mode

rongpengzhang committed:
Temp

rongpengzhang committed:
Implement the method to address the coupling between OA Mixer and VRF VAV coil simulation

Message

rongpengzhang committed:
Reorganize the codes

rongpengzhang committed:
Refine the algorithm to determine variable Te/Tc

Take into account the load introduced by the outdoor air

rongpengzhang committed:
Modification on the IU coil

Clean up the IU coil calculations

Created ControlVRF_FluidTCtrl method just for the VRF_FluidTCtrl model,
instead of setting conditions in the ControlVRF which also serves
VRF_SysCurve model.

rongpengzhang committed:
Extend the codes to allow multiple thermostat priority control type

rongpengzhang committed:
Update Unit Test

Myoldmopar commented on idd/Energy+.idd.in:
```diff
@@ -54105,6 +54105,26 @@ AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl,
+       \type object-list
+       \object-list ScheduleNames
+       \note this field is required if Master Thermostat Priority Control Type is Scheduled
+       \note Schedule values of 0 denote cooling, 1 for heating, and all other values disable the system
```
Ummmmmm, OK.  So this can't go in the bug fix release then.  

@rongpengzhang Is there a way to extract out just the bug fix portion without these idd changes?  Or are they coupled?  

FYI @mjwitte

Myoldmopar commented on src/EnergyPlus/DXCoils.cc:
```diff
@@ -6061,7 +6061,7 @@ namespace DXCoils {
 					TempSize = DXCoil( DXCoilNum ).RatedTotCap( Mode );
 					SizingString = DXCoilNumericFields( DXCoilNum ).PerfMode( Mode ).FieldNames( FieldNum ) + " [W]";
-					CalcVRFCoilCapModFac( 0, _, CompName, FinalZoneSizing ( CurZoneEqNum ).DesCoolCoilInTemp, _, _, _, DataTotCapCurveValue);
+					CalcVRFCoilCapModFac( 0, _, CompName, ZoneSizingRunDone ? FinalZoneSizing ( CurZoneEqNum ).DesCoolCoilInTemp : 26, _, _, _, DataTotCapCurveValue);
```
@rongpengzhang I appreciate the shorthand here, and it's always a tough balance for me between conciseness and clarity, but in this case I'd prefer to have the `ZoneSizingRunDone ? FinalZoneSizing ( CurZoneEqNum ).DesCoolCoilInTemp : 26` portion pulled into a separate line/variable above.

Myoldmopar commented on src/EnergyPlus/DXCoils.cc:
```diff
@@ -14292,7 +14292,7 @@ Label50: ;
 		return SHR;
 	}
-
+	
```
You might see if Visual Studio has a setting for automatically deleting any trailing whitespace when saving a file.  @mjwitte do you know of this setting?

mjwitte replied:
I haven't found that setting in Visual Studio (delete trailing whitespace).

mitchute replied:
[CodeMaid](http://www.codemaid.net/) works great for my install of VS 2012, but doesn't load for 2013 Express. Maybe I just don't have something setup correctly. If anyone tries it and gets it to work, feel free to pass it along (in a separate thread).

Myoldmopar commented on src/EnergyPlus/DXCoils.cc:
```diff
@@ -14350,7 +14352,7 @@ Label50: ;
 		Real64 VolFlowperRatedTotCap; // Air volume flow rate divided by rated total cooling capacity [m3/s-W] (adjusted for bypass)
 		Real64 TotCap; // gross total cooling capacity at off-rated conditions [W]
-		Real64 InletAirWetBulbC; // wetbulb temperature of inlet air [C]
+		// Real64 InletAirWetBulbC; // wetbulb temperature of inlet air [C]
```
If it isn't used now, go ahead and delete it instead of just commenting.

Myoldmopar commented:
@rongpengzhang I left a few comments on the code, but I don't have much else.  Given that you have input changes, this can't go in the bug fix release, so I won't be looking at this again for at least a week.  

If you can extract out the bug fix without changing inputs, please issue a new PR right away with the fix.  If you can't, then can you update the wrap-up ticket #5355 to say it won't be fixed for the bug fix release, and also re-assign the original VRF ticket #5252 to [EnergyPlus 8.5.0](https://github.com/NREL/EnergyPlus/milestones/EnergyPlus%208.5.0)?  Thanks.

rongpengzhang commented:
@Myoldmopar Thanks for the reivew and comments. I just finished the coding of this issue yesterday night. I planned to invite you or Mike or Richard when it passed all the tests. But it is great that you have reviewed it.

There is no problem to delete the IDD change part. This can be separated from this issue, and be merged when I develop the HR algorithm.

I will finish an updated version in hours.

Myoldmopar commented:
That's excellent news @rongpengzhang; I'll look forward to doing some testing of it when you get the updated version without IDD changes pushed up.

rongpengzhang committed:
Updated Version for Bug Fix Release

Remove the IDD updates
Address Edwin's comments and build warnings

rongpengzhang committed:
Merge remote-tracking branch 'refs/remotes/origin/develop' into #5252_TinDiscrepancyNewVRFModel

rongpengzhang committed:
Merge branch '#5252_TinDiscrepancyNewVRFModel' of https://github.com/NREL/EnergyPlus into #5252_TinDiscrepancyNewVRFModel

rongpengzhang committed:
Updated version for bug fix release

It seems the IDD change was not moved in the previous commit. Maybe the
previous commit was considered less updated by GitHub. Anyway, do it
again

rongpengzhang commented:
@Myoldmopar Updated version is done.

rongpengzhang committed:
Deleting trailing whitespace in IDD

Myoldmopar commented:
@rongpengzhang Are you able to identify where you might have changed the code so that a `using Fans::Fan` isn't needed anymore?  That's the build warning from above.  I can build it on Ubuntu and reproduce, but maybe you'll already see it.

rongpengzhang committed:
Address warning message

rongpengzhang commented:
Hi @Myoldmopar I found the codes that I changed so that the "using Fans::Fan" isn't needed anymore. I think the warning message will be gone.

rongpengzhang commented:
@Myoldmopar I created a new commit in the branch, and it goes to PR #5378, not this PR. Maybe because you closed this PR

rongpengzhang commented:
Hi @mjwitte, can you please tell me how I can see the CI test result for a PR that has been merged and closed, such as this one? I got the "404 File not found" if I directly click the "build status" button. It seems that you can get access to that.

Myoldmopar commented:
@rongpengzhang Can you change the title of this to something a user will understand so that it comes out nicely in the ChangeLog?  Thanks.

rongpengzhang commented:
@Myoldmopar How about _Add OA mixer in terminal unit of the VRF-FluidTCtrl model_?

Myoldmopar commented:
Thanks @rongpengzhang
